src/librustc_metadata/creader.rs

```diff
@@ -76,6 +76,21 @@ impl<'a> LoadError<'a> {
     }
 }
 
+/// A pointer to `CrateMetadata` that can also give access to whole crate store when necessary.
+#[derive(Clone, Copy)]
+crate struct CrateMetadataPtr<'a> {
```

Comment:
Would you be fine with calling this `CrateMetadataRef`? Or `CrateMetadataHandler` etc.?
Too used to "ptr" meaning raw pointer.